tests/benchmarks/latency_measure/src/timing_info.h

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2017 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _TIMING_INFO_H_
+#define _TIMING_INFO_H_
+
+#include <timestamp.h>
+
+#ifdef CONFIG_NRF_RTC_TIMER
+#include <soc.h>
+
+/* To get current count of timer, first 1 need to be written into
+ * Capture Register and Current Count will be copied into corresponding
+ * current count register.
+ */
+#define TIMING_INFO_PRE_READ()        (NRF_TIMER2->TASKS_CAPTURE[0] = 1)
+#define TIMING_INFO_OS_GET_TIME()     (NRF_TIMER2->CC[0])
+
+#elif CONFIG_SOC_SERIES_MEC1501X
+#include <soc.h>
+
+#define TIMING_INFO_PRE_READ()
+#define TIMING_INFO_OS_GET_TIME()     (B32TMR1_REGS->CNT)
+
+#else
+#define TIMING_INFO_PRE_READ()
+#define TIMING_INFO_OS_GET_TIME()      (k_cycle_get_32())
+#endif
+
+/******************************************************************************/
+/* NRF RTC TIMER runs ar very slow rate (32KHz), So in order to measure
+ * Kernel starts a dedicated timer to measure kernel stats.
+ */
+#ifdef CONFIG_NRF_RTC_TIMER
+#define NANOSECS_PER_SEC (1000000000)
+#define CYCLES_PER_SEC   (16000000/(1 << NRF_TIMER2->PRESCALER))
```

Comment:
FWIW: SysTick runs at full CPU cycle speed (64 MHz in this case) and requires no probing, any reason you're not using that instead?

Reply:
This is copied from `tests/benchmarks/timing_info/src/timing_info.h` directly, so it just follows what is being done there.